__tests__/e2e/helpers/createForm.ts

```diff
@@ -614,3 +294,368 @@ const addCollaborators = async (
+      )
+      await fillDropdown(
+        page,
+        page.getByRole('textbox', { name: 'Shape' }),
+        field.ratingOptions.shape,
+      )
+      break
+    case BasicField.Table:
+      await page.getByLabel('Minimum rows').fill(String(field.minimumRows))
+      if (field.addMoreRows) {
+        await page.getByText('Allow respondent to add more rows').click()
+        if (field.maximumRows) {
+          await page
+            .getByLabel('Maximum rows allowed')
+            .fill(String(field.maximumRows))
+        }
+      }
+      // First table option
+      for (let i = 0; i < field.columns.length; i++) {
+        const col = field.columns[i]
+        if (i !== 0) {
+          await page.getByRole('button', { name: 'Add column' }).click()
+        }
+        await page.getByLabel(`Column ${i + 1}`).fill(col.title)
+        await page.getByLabel('Column type').nth(i).click()
+        await page
+          .getByRole('option', {
+            name: BASICFIELD_TO_DRAWER_META[col.columnType].label,
+          })
+          .click()
+        if (!col.required) {
+          await page.getByText('Required').nth(i).click()
+        }
+        if (col.columnType === BasicField.Dropdown) {
+          await page
+            .locator(`[id="columns\\.${i}\\.fieldOptions"]`)
+            .fill(col.fieldOptions.join('\n'))
+        }
+      }
+      break
+  }
+}
+
+/** Adds all prescribed MyInfo fields to the form.
+ * Precondition: page must be currently on the admin builder page for the form.
+ * @param {Page} page Playwright page
+ * @param {E2eFieldMetadata} field the form field to create
+ * @param {MyInfoAttribute} attr the MyInfo attribute to be added
+ */
+const addMyInfoField = async (
+  page: Page,
+  field: E2eFieldMetadata,
+  attr: MyInfoAttribute,
+): Promise<void> => {
+  const label = MYINFO_FIELD_TO_DRAWER_META[attr].label
+
+  await page.getByRole('tab', { name: 'MyInfo' }).click()
+  await page.getByRole('button', { name: label, exact: true }).click()
+}
+
+/** Goes to settings page and adds settings, and toggle form to be open.
+ * Precondition: is already on the Create page.
+ * @param {Page} page Playwright page
+ * @param {E2eFieldMetadata[]} formFields the form fields used to create the form
+ * @param {E2eLogic[]} formLogics the form logic to create
+ */
+const addLogics = async (
+  page: Page,
+  formFields: E2eFieldMetadata[],
+  formLogics: E2eLogic[],
+) => {
+  if (formLogics.length === 0) return
+
+  // Navigate to the logic tab.
+  await page.getByRole('button', { name: 'Add logic' }).click()
+
+  for (const logic of formLogics) {
+    // The 0th button called 'Add logic' is the sidebar tab nav button
+    await page.getByRole('button', { name: 'Add logic' }).nth(1).click()
+
+    // Add logic conditions
+    for (let i = 0; i < logic.conditions.length; i++) {
+      const { field, state, value } = logic.conditions[i]
+
+      if (i) await page.getByRole('button', { name: 'Add condition' }).click()
```

Comment:
Nit: is `if (i)` trying to skip the first logic? Would prefer if this skip is more explicit through `if (i > 0)` or even `if (i !== 0)`.